design/cdk-bootstrap.md

```diff
@@ -0,0 +1,637 @@
+  This role will be assumable by any principal from the account(s) passed by the `--trust` option.
+
+* An IAM role, called the **Deployment Action Role**,
+  that will be assumed when executing the CloudFormation deployment actions
+  (CreateChangeSet and ExecuteChangeSet).
+  It is also assumable by any principal from the account(s) passed by the `--trust` option.
+
+* An IAM role, called the **CloudFormation Execution Role**,
+  that will be used to perform the actual CFN stack deployment in the continuous delivery pipeline to this environment.
+  It is assumable *only* by the CloudFormation service principal
+  (this is for security reasons, as this role will have, necessarily, very wide permissions).
+  It will not have any inline policies,
+  but will instead have the Managed Policies attached that the user selected when prompted (see above).
+
+#### Physical resource names
+
+All of the above resources will be created with well-defined physical names -
+none of them will rely on automated CloudFormation naming.
+This is so that the other stages of CDK synthesis,
+like asset resolution, can rely on the concrete names
+(there is no reliable way to transfer this kind of information in an automated way across regions and/or accounts).
+This also allows for more fine-grained permissions -
+for instance, the continuous delivery pipeline needs to grant the **Deployment Action Role**
+permissions to read from the pipeline's bucket
+(to read the artifact that contains the CFN template to deploy);
+this way, it can add a well-defined name to the resource policy of the artifact bucket,
+instead of granting those permissions to all principals in the account.
+
+The naming scheme will include the following elements in order to minimize the chance of name collisions:
+
+* The region we're bootstrapping in.
+* The account ID we're bootstrapping in.
+* The type of the resource (file assets bucket, Docker assets repository, etc.).
+
+### `--trust`
+
+We will add a new command-line option called `--trust` to the `bootstrap` command.
+Its value is a list of AWS account IDs:
+
+```shell
+$ cdk bootstrap \
+    [--trust accountId[,otherAccountId]...] \
+    [--cloudformation-execution-policies policyName[,otherPolicyName]...] \
+    aws://account/region
+```
+
+If a user wants to add new trusted account(s) to an existing bootstrap stack,
+they have to specify all of the accounts they want to trust in the `--trust` option,
+not only the one being newly added -
+otherwise, the previously trusted account(s) will be removed.
+
+We will also add a another option,
+`--cloudformation-execution-policies`,
+that allows you to pass a list of managed policy ARNs on the command line to attach to the
+**CloudFormation Execution Role**.
+
+### Removing existing customization options
+
+The existing customization options: `--bootstrap-bucket-name` and `--bootstrap-kms-key-id` will be removed.
```

Comment:
I removed them because you explicitly asked me to in previous comments ([example](https://github.com/aws/aws-cdk/pull/4461#discussion_r338939147), and that whole thread).

Should I add them back in?

Reply:
just the stack name seems like something that we can continue to support without implications to our design.

Reply:
Sorry, I don't understand your comments here...

We're discussing here the following line in the doc:

> The existing customization options: `--bootstrap-bucket-name` and `--bootstrap-kms-key-id` will be removed.

Your previous comment was (emphasis mine):

> I think **--bootstrap-bucket-name** is something we can still support. We don't depend on it in the CI/CD story for anything.

Are you saying you meant `--toolkit-stack-name` above?

Reply:
YES! Sorry. My bad! Brain fart. Let's keep `toolkit-stack-name`

Reply:
`--toolkit-stack-name` is already explicitly mentioned as being kept here: https://github.com/aws/aws-cdk/pull/4461/files#diff-a8df670df96ab0a6863933b563668ba2R100